yarpl/include/yarpl/observable/ObservableOperator.h

```diff
@@ -445,8 +461,44 @@ class FromPublisherOperator : public Observable<T> {
   explicit FromPublisherOperator(OnSubscribe function)
       : function_(std::move(function)) {}
 
-  void subscribe(Reference<Observer<T>> observer) override {
-    function_(std::move(observer));
+ private:
+  class PublisherObserver : public Observer<T> {
```

Comment:
I could not understand why did we need this class. Is it for just to eliminate calling onSubscribe?

Reply:
Exactly. As I am following the Java RX, we also don't want the application code to call onSubscribe in the lambdas. I believe they choose this behavior because it adds more states the subscriber can be in when you get it as parameter for your lambda:
Observable::create([](Subscriber s) { /*its already subscribed, so I don't have to worry about onSubscribe*/ });

PublisherObserver just swallows the onSubscribe in release mode, but it errors out in the debug mode.